kythe/go/platform/tools/indexer_wrapper/main.go

```diff
@@ -0,0 +1,122 @@
+// Binary indexer_wrapper reads compilation units from the provided kzip(s) and
+// indexes them using language-specific indexers. Entry protos are written to
+// stdout.
+//
+// TODO(https://github.com/kythe/kythe/pull/3339): after the analyzer driver
+// interface is finalized, update this binary to use it rather than creating a
+// temporary kzip file for each indexer invocation.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"kythe.io/kythe/go/platform/analysis"
+	"kythe.io/kythe/go/platform/analysis/driver"
+	"kythe.io/kythe/go/platform/analysis/local"
+	"kythe.io/kythe/go/platform/delimited"
+
+	apb "kythe.io/kythe/proto/analysis_go_proto"
+)
+
+// delegatingAnalyzer provides an implementation of the CompilationAnalyzer
+// interface that forwards to a set of language-specific analyzers.
+type delegatingAnalyzer struct {
+	analyzers map[string]analysis.CompilationAnalyzer
+}
+
+// binPath returns the absolute path to an indexer given its name.
+func binPath(binname string) string {
+	return filepath.Join(*kytheDir, "indexers", binname)
+}
+
+func NewDelegatingAnalyzer(fetcher analysis.Fetcher) *delegatingAnalyzer {
```

Comment:
Unless you're thinking about pulling this out into a library, I recommend unexporting it. It really doesn't matter that much in package main, but I find it helpful as a reader (seeing "unexported" is good signal).

Reply:
:+1: